config/config.go

```diff
@@ -1126,7 +1128,13 @@ func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return err
 	}
 	if c.Region == "" {
-		return fmt.Errorf("EC2 SD configuration requires a region")
+		sess := session.Must(session.NewSession())
```

Comment:
Sounds like this could panic if something goes wrong. Better to check potential errors?